bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/SoulissHandlerFactory.java

```diff
@@ -0,0 +1,179 @@
+import org.openhab.binding.souliss.handler.SoulissT16Handler;
+import org.openhab.binding.souliss.handler.SoulissT18Handler;
+import org.openhab.binding.souliss.handler.SoulissT19Handler;
+import org.openhab.binding.souliss.handler.SoulissT1AHandler;
+import org.openhab.binding.souliss.handler.SoulissT22Handler;
+import org.openhab.binding.souliss.handler.SoulissT31Handler;
+import org.openhab.binding.souliss.handler.SoulissT41Handler;
+import org.openhab.binding.souliss.handler.SoulissT42Handler;
+import org.openhab.binding.souliss.handler.SoulissT51Handler;
+import org.openhab.binding.souliss.handler.SoulissT52Handler;
+import org.openhab.binding.souliss.handler.SoulissT53Handler;
+import org.openhab.binding.souliss.handler.SoulissT54Handler;
+import org.openhab.binding.souliss.handler.SoulissT55Handler;
+import org.openhab.binding.souliss.handler.SoulissT56Handler;
+import org.openhab.binding.souliss.handler.SoulissT57Handler;
+import org.openhab.binding.souliss.handler.SoulissT61Handler;
+import org.openhab.binding.souliss.handler.SoulissT62Handler;
+import org.openhab.binding.souliss.handler.SoulissT63Handler;
+import org.openhab.binding.souliss.handler.SoulissT64Handler;
+import org.openhab.binding.souliss.handler.SoulissT65Handler;
+import org.openhab.binding.souliss.handler.SoulissT66Handler;
+import org.openhab.binding.souliss.handler.SoulissT67Handler;
+import org.openhab.binding.souliss.handler.SoulissT68Handler;
+import org.openhab.binding.souliss.handler.SoulissTopicsHandler;
+import org.openhab.binding.souliss.internal.protocol.SoulissBindingNetworkParameters;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SoulissHandlerFactory} is responsible for creating things and thingGeneric
+ * handlers. It fire when a new thingGeneric is added.
+ *
+ * @author Tonino Fazio - Initial contribution
+ * @author Luca Calcaterra - Refactor for OH3
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.souliss", service = ThingHandlerFactory.class)
+public class SoulissHandlerFactory extends BaseThingHandlerFactory {
+    private final Logger logger = LoggerFactory.getLogger(SoulissHandlerFactory.class);
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (thingTypeUID.toString().equals(GATEWAY_THING_TYPE.getAsString().toLowerCase())) {
```

Comment:
This if else statement is very long with around 19 `else if`'s... Can this be written using a switch statement?

Reply:
Pending. No time to work on this..

Reply:
No problem, I'll close this down since it is a V1 updated binding and you have already done a great job updating it.